src/auth/digest/Config.cc

```diff
@@ -807,8 +808,15 @@ Auth::Digest::Config::decode(char const *proxy_auth, const char *aRequestRealm)
         switch (t) {
         case DIGEST_USERNAME:
             safe_free(username);
-            if (value.size() != 0)
-                username = xstrndup(value.rawBuf(), value.size() + 1);
+            if (value.size() != 0) {
+                const auto v = value.termedBuf();
+		if (utf8 && !isValidUtf8String(v, v + value.size())) {
```

Comment:
Please remove tabs from code indentation. More lines below too

```suggestion
        if (utf8 && !isValidUtf8String(v, v + value.size())) {
```